verify: Upgrade verus to official release-2025-11-16

ziqiaozhou opened the pull request:
* Update verus builtin crate names
* Update some annotations since latest verus handles const items in a slightly different ways.
* Add verus_verify for all Impl item since it is ignored if not marked as verus_verify in latest verus.
* Update tool to download prebuilt verus

stefano-garzarella commented on scripts/vinstall.sh:
```diff
@@ -5,27 +5,18 @@
+# Install x86_64-unknown-none target for verus-compatible Rust version
+export RUSTUP_TOOLCHAIN=$VERUS_RUST_VERSION
+rustup toolchain install $RUSTUP_TOOLCHAIN
+rustup target add x86_64-unknown-none
```
What about this?

```suggestion
rustup target add x86_64-unknown-none --toolchain $RUSTUP_TOOLCHAIN
```

ziqiaozhou commented on scripts/vinstall.sh:
```diff
@@ -5,27 +5,18 @@
-RUSTUP_TARGET="x86_64-unknown-none"
-rustup target add "$RUSTUP_TARGET"
+VERISMO_REV=ffee0c9
+VERUS_RUST_VERSION=1.91
```
Yes. Just to ensure it works well in a clean env and in the CI #873, I tried it at https://github.com/MSRSSP/svsm/pull/21 where I use verus-workflow branch as main branch in my repo. Since this PR is from a branch with verus keyworkd, it will trigger the workflow.

If #873 got merged into main, svsm repo should see "Verification" workflow under actions and can trigger it manually, like https://github.com/MSRSSP/svsm/actions/runs/19556358785

stefano-garzarella commented on scripts/vinstall.sh:
@ziqiaozhou I re-opened this, since I don't see this addressed. It's a minor thing, but it's what we usually do in this repo, so we can identify fast what subsystem a commit affect. That will be nice for backports and bisection.

ziqiaozhou commented:
> > I think it is due to missing rust toolchain 1.91-x86_64-unknown-linux-gnu (if you run verus directly, it should throw out that error).
> 
> edit: ~How can I do that?~
> 
> ```
> $ verus --version
> verus: required rust toolchain 1.91.0-x86_64-unknown-linux-gnu not found
> ```
> 
> The `VERUS_RUST_VERSION` is wrong, see [#868 (comment)](https://github.com/coconut-svsm/svsm/pull/868#discussion_r2545073553)
> 
> > The upated version now will install verus from a prebuilt based on your os and arch since building verus is slow. Due to that reason, the script will not automatically download Rust toolchain 1.91. I updated it to install the toolchain automatically.
> 
> edit: ~I still have the panic here :-(~ fixed with [#868 (comment)](https://github.com/coconut-svsm/svsm/pull/868#discussion_r2545073553)
> 
> BTW it's strange to have a panic in that case, why not an error?

Although it may not matter in the updated vinstall.sh. I updated the cargo-v tool version to propagate proper error from verus tool. This is a good suggestion.
